airflow/www/views.py

```diff
@@ -2554,6 +2552,20 @@ def grid(self, dag_id, session=None):
             'dag_runs': encoded_runs,
         }
 
+        dag_details_url = ''
+        tasks_url = ''
+        # This is a workround to use try/except if url_for() throws an error
```

Comment:
During tests `url_for` was throwing an error, so I moved the logic into `views.py` but I'm all ears on a better way to handle this.
It would be nice to have a sort of `url_for(RestAPI)` function.

Reply:
The api init step is skipped in those test. Try this instead:

```
--- a/tests/www/views/test_views_log.py
+++ b/tests/www/views/test_views_log.py
@@ -56,7 +56,12 @@ def backup_modules():
 @pytest.fixture(scope="module")
 def log_app(backup_modules):
     @dont_initialize_flask_app_submodules(
-        skip_all_except=["init_appbuilder", "init_jinja_globals", "init_appbuilder_views"]
+        skip_all_except=[
+            "init_appbuilder",
+            "init_jinja_globals",
+            "init_appbuilder_views",
+            "init_api_connexion",
+        ]
     )
     @conf_vars({('logging', 'logging_config_class'): 'airflow_local_settings.LOGGING_CONFIG'})
     def factory():
```